native-src/android/app/src/main/java/com/telerik/pushplugin/PushPlugin.java

```diff
@@ -124,15 +119,17 @@ public void onMessageReceived(RemoteMessage message) {
         RemoteMessage.Notification notif = message.getNotification();
         Log.d(TAG, "New Push Message: " + data);
         Log.d(TAG, "Msg notification: " + notif);
+        if (notif != null) {
+            Log.d(TAG, "Notification body: " + notif.getBody());
+        }
+
+        boolean isCallbackRegistered = onMessageReceivedCallback != null;
+        Log.d(TAG, "Callback is registered: " + isCallbackRegistered);
 
-        // If the application has the callback registered and Plugin is active
-        // execute the callback. Otherwise, create new notification in the notification bar of the user.
-        if (onMessageReceivedCallback != null && isActive) {
+        // If the application has the callback registered
+        // execute the callback. Otherwise, let the OS handle the notification.
+        if (isCallbackRegistered) {
             executeOnMessageReceivedCallback(data, notif);
-        } else {
-            Log.d(TAG, "Creating our own notification in tray...");
-            Context context = getApplicationContext();
-            NotificationBuilder.createNotification(context, data);
```

Comment:
This does ignore the notification we used to build, but I think we should remove the now unused code from the plugin. The NotificationBuilder class, as well as the code which deals with the extras it set as "pushData" and maybe other things too.